scripts/publish-local-to.mjs

```diff
@@ -10,108 +10,128 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
-import { resolve, relative } from "path";
+import { resolve, relative, basename } from "path";
 import { readFileSync } from "fs";
 import {
   getSdks,
   highlightLogVars,
   logger,
+  repoRoot,
   runWithArg,
-  sh,
+  sh as realSh,
   shResult,
 } from "./script-runner.mjs";
 
-let absDependentDir;
-let dependentName;
+// for dry runs
+const fakeSh = (cmd, args, opts = {}) => {
+  const cwd = relative(repoRoot, opts.cwd || "") || "./";
+  logger.log.hl`would run ${cwd} > ${basename(cmd)} ${args.join(" ")}`;
+};
+
+const bullets = (iter) => ["", ...iter].join("\n • ");
+
+async function publishLocalTo({ dryRun }, dependents) {
+  const sh = dryRun ? fakeSh : realSh;
 
-async function publishLocalTo(dependent) {
-  const workDir = process.cwd();
   const yalc = resolve(await shResult("npm", ["bin"]), "yalc");
   try {
     await shResult(yalc, ["--version"]);
   } catch (e) {
-    throw new Error(`Could not find "yalc" in npm path: ${e.message}`);
+    throw new Error(
+      `Could not find "yalc" in npm path: ${e.message}. Install yalc globally with yarn or npm to proceed.
+(Sorry, 'npx yalc' doesn't work in this use case.`
+    );
   }
 
   const sdks = await getSdks();
-  const sdkPackages = new Set(sdks.map((sdkPart) => sdkPart.pkg.name));
-  const unlinkedSdkPackages = new Set();
 
-  if (dependent) {
-    const absDependentDir = resolve(workDir, dependent);
+  logger.log.hl`Publishing ${sdks.map((s) => s.shortName).join(", ")}`;
+  for (const { cwd } of sdks) {
+    await sh(yalc, ["publish", "--quiet", "--changed"], { cwd, silent: true });
+  }
+
+  for (const dependent of dependents) {
     const usedSdkPackages = new Set();
+    const unlinkedSdkPackages = new Set();
     const yalcConfigDir = await shResult(yalc, ["dir"], {
-      cwd: absDependentDir,
+      cwd: dependent.dir,
     });
-    const yalcInstallations = JSON.parse(
-      readFileSync(resolve(yalcConfigDir, "installations.json"))
-    );
-
-    logger.log(
-      "Checking if %s has already yalc-added SDK packages",
-      dependentName
-    );
+    let yalcInstallations = {};
     try {
-      const found = JSON.parse(
-        await shResult("npm", ["explain", "--json", ...sdkPackages], {
-          cwd: absDependentDir,
-        })
+      yalcInstallations = JSON.parse(
+        readFileSync(resolve(yalcConfigDir, "installations.json"))
```

Comment:
Stop relying on `npm` or `yarn` to give us reliable JSON and just read yalc installations instead. Ideally, yalc would be able to spit out JSON from the `installations show` command, but it doesn't, so we look directly at the installations file.